### PR TITLE
6 ticket

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -109,4 +109,50 @@ describe("All requests", () => {
...
+                        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+                        expect(comment).toHaveProperty("votes", expect.any(Number));
+                        expect(comment).toHaveProperty("created_at", expect.any(String));
+                        expect(comment).toHaveProperty("author", expect.any(String));
+                        expect(comment).toHaveProperty("body", expect.any(String));
+                        expect(comment).toHaveProperty("article_id", expect.any(Number));
+                    });
+                });
+        });
+        it("200: if articles contain this params  ", () => {
```
Comment (NRMSMITH) [inline __tests__/app.test.js:133]: Please finish your descriptions

File: __tests__/app.test.js [hunk 2]
```diff
@@ -109,4 +109,50 @@ describe("All requests", () => {
...
+        });
+        test("400: and msg of Invalid request!", () => {
+            return request(app)
+                .get("/api/articles/mumboJumbo/comments")
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toBe("Invalid request!");
+                });
+        });
+        test("404:  and msg of Not found!", () => {
```
Comment (NRMSMITH) [inline __tests__/app.test.js:149]: These descriptions need to be more descriptive - i.e. '404: reponds with a msg of Not Found when the article id is valid but does not exist'.

File: controllers/articles.controller.js [hunk 3]
```diff
@@ -17,3 +22,19 @@ exports.getAllArticles = (req, res, next) => {
...
+
+    const promises = [selectCommentsByArticleId(article_id)];
+
+    if (article_id) {
+        promises.push(checkArticleExist(article_id));
+    }
+
+    return Promise.all(promises)
+        .then((comments) => {
+            res.status(200).send({ comments: comments[0] });
```
Comment (NRMSMITH) [inline controllers/articles.controller.js:37]: The comments[0] should really only happen in the model 😄

**Review by NRMSMITH — APPROVED** [review]
Really tiny tweaks - please correct them and then merge :)

File: controllers/articles.controller.js (hunk 3 above)
Comment (Hassen-Ahmed) [reply controllers/articles.controller.js:37]: Thank you so much and I will fix and merge this.

Comment (Hassen-Ahmed) [reply controllers/articles.controller.js:37]: About the last comment you gave me, it is a Promise.all() that's why I did that and how could I make it happen in side model?? Ty.

Comment (Hassen-Ahmed) [reply controllers/articles.controller.js:37]: > The comments[0] should really only happen in the model 😄

About the last comment you gave me, it is a Promise.all() that's why I did that and how could I make it happen in side model?? Ty.

**Pull request closed without merging**